File: src/main/java/com/lessmarkup/interfaces/system/LanguageCache.java
package com.lessmarkup.interfaces.system;

import com.lessmarkup.interfaces.cache.CacheHandler;
import java.util.List;
import java.util.Map;

public interface LanguageCache extends CacheHandler {
    String getCurrentLanguageId();
    default String getTranslation(String id, String moduleType, boolean throwIfNotFound) {
        return getTranslation(getCurrentLanguageId(), id, moduleType, throwIfNotFound);
    }
    default String getTranslation(String id, String moduleType) {
        return getTranslation(id, moduleType, true);
    }
    String getTranslation(String languageId, String id, String moduleType, boolean throwIfNotFound);
    default String getTranslation(String languageId, String id, String moduleType) {
        return getTranslation(languageId, id, moduleType, true);
    }
    List<Language> getLanguages();
}

File: src/main/scala/com/lessmarkup/interfaces/system/LanguageCache.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0.
 * If a copy of the MPL was not distributed with this file, You can obtain one at
 * http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.interfaces.system

import com.lessmarkup.interfaces.cache.CacheHandler

trait LanguageCache extends CacheHandler {
  def getCurrentLanguageId: Option[String]

  def getTranslation(id: String, moduleType: Option[String], throwIfNotFound: Boolean): Option[String] = {
    getTranslation(getCurrentLanguageId, id, moduleType, throwIfNotFound)
  }

  def getTranslation(id: String, moduleType: Option[String]): Option[String] = {
    getTranslation(id, moduleType, throwIfNotFound = true)
  }

  def getTranslation(languageId: Option[String], id: String, moduleType: Option[String], throwIfNotFound: Boolean): Option[String]

  def getTranslation(languageId: Option[String], id: String, moduleType: Option[String]): Option[String] = {
    getTranslation(languageId, id, moduleType, throwIfNotFound = true)
  }

  def getLanguages: List[Language]
}